src/js/components/Drop/DropContainer.js

```diff
@@ -202,7 +235,29 @@ const DropContainer = forwardRef(
             top =
               targetRect.top + targetRect.height / 2 - containerRect.height / 2;
           }
-          container.style.left = `${left}px`;
+
+          let containingBlock;
+          let containingBlockRect;
+          // dropOptions was created to preserve backwards compatibility
+          if (dropOptions?.checkContainingBlock) {
+            // return the containing block for absolute elements or `null`
+            // for fixed elements
+            containingBlock = getContainingBlock(container);
+            containingBlockRect = containingBlock?.getBoundingClientRect();
+          }
+
+          // compute viewport offsets
+          const viewportOffsetLeft = containingBlockRect?.left ?? 0;
+          const viewportOffsetTop = containingBlockRect?.top ?? 0;
+          const viewportOffsetBottom =
+            containingBlockRect?.bottom ?? windowHeight;
+
+          const containerOffsetLeft = containingBlock?.scrollLeft ?? 0;
+          const containerOffsetTop = containingBlock?.scrollTop ?? 0;
+
+          container.style.left = `${left - viewportOffsetLeft +
+            containerOffsetLeft}px`;
```

Comment:
I'm wondering if we can structure this code so it is a bit more contained and is only run when `dropOptions.checkContainingBlock = true`. For example:

```
if (dropOptions?.checkContainingBlock && getContainingBlock(container)) {
  ...
} else {
  ...
}
```

Reply:
imo, `containingBlockRect = containingBlock?.getBoundingClientRect();` is the expensive operation and it only gets run if the option is enabled.

All of the nullish coalescing in the extra variables that depend on `containingBlockRect` is not a performance hit. Maybe this is just a stylistic suggestion but I prefer the current style, defining each value in a single `const` declaration

Reply:
No performance concerns here, rather I just wanted to organize the code this way for readability. Since it is an opt in behavior and not the default it would be nice (but not critical) to have it separated a bit more in the code.